helm-command.el

```diff
@@ -264,7 +255,11 @@ Unlike regular `M-x' emacs vanilla `execute-extended-command' command,
 the prefix args if needed, are passed AFTER starting `helm-M-x'.
 
 You can get help on each command by persistent action."
-  (interactive (list current-prefix-arg (helm-M-x-read-extended-command)))
+  (declare (interactive-only command-execute))
+  (interactive
+   (progn
+     (setq helm-M-x-prefix-argument current-prefix-arg)
```

Comment:
Even simpler if perhaps using something like this which avoid the unused arg ARG at compile time
and is simpler IMO (correct me if I am wrong, I am tired tonight).

```elisp
  (interactive
   (progn
     current-prefix-arg
     (list current-prefix-arg (helm-M-x-read-extended-command))))
  (setq helm-M-x-prefix-argument arg)
```